### PR TITLE
Coq 8.7.1

### PR DESCRIPTION
**Pull request by RalfJung**
I hope I got the sha512 checksum syntax right; can't test that with pre-2.0 opam, it seems.

Cc @maximedenes

### CODE REVIEW

File: packages/coq/coq.8.7.1/opam [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  "%{share}%/texmf/tex/latex/misc/coqdoc.sty"
+  "%{share}%/emacs/site-lisp/coq-font-lock.el"
+  "%{share}%/emacs/site-lisp/coq-inferior.el"
+  "%{share}%/emacs/site-lisp/gallina-db.el"
+  "%{share}%/emacs/site-lisp/gallina.el"
+  "%{share}%/emacs/site-lisp/gallina-syntax.el"
+  ]
+]
+depends: [
+  "ocamlfind"
```
Comment (jpdeplaix) [inline packages/coq/coq.8.7.1/opam]: Could you add the ```build``` attribute to ```ocamlfind``` ? It seems that only the ```ocamlfind``` tool is used and not the library.
Is it also the case for ```campl5``` ?

File: packages/coqide/coqide.8.7.1/opam [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "-datadir" "%{share}%/coq"
+    "-usecamlp5"
+    "-camlp5dir" "%{lib}%/camlp5"
+  ]
+  [make "-j%{jobs}%" "coqide-files"]
+  [make "-j%{jobs}%" "coqide-opt"]
+]
+remove: ["rm" "-R" "%{lib}%/coq/ide" "%{doc}%/FAQ-CoqIde"]
+depends: [
+  "camlp5"
```
Comment (jpdeplaix) [inline packages/coqide/coqide.8.7.1/opam]: Same question here.

File: packages/coq/coq.8.7.1/opam (hunk 1 above)
Comment (ejgallego) [reply packages/coq/coq.8.7.1/opam]: ocamlfind/camlp5 is necessary for those people wanting to develop against Coq. I am not sure what is the OPAM policy on `-dev` libraries [in the Debian sense], but it is required for Coq that people doing `opam install coq` are able to build Coq plugins afterwards.

File: packages/coqide/coqide.8.7.1/opam (hunk 2 above)
Comment (ejgallego) [reply packages/coqide/coqide.8.7.1/opam]: Same comment than above.

File: packages/coq/coq.8.7.1/opam (hunk 1 above)
Comment (jpdeplaix) [reply packages/coq/coq.8.7.1/opam]: The ```build``` attribute just tells the OPAM solver to not rebuild the package in the dependency in question is upgraded. Nothing is changed for users or plugin developers.

Comment (ejgallego) [reply packages/coq/coq.8.7.1/opam]: I see thanks, so then the answer is that the attribute should be added to ocamlfind, but not to `camlp5`.

File: packages/coqide/coqide.8.7.1/opam (hunk 2 above)
Comment (ejgallego) [reply packages/coqide/coqide.8.7.1/opam]: Coq links to camlp5 so indeed if campl5 is updated Coq has to be rebuilt.

File: packages/coq/coq.8.7.1/opam (hunk 1 above)
Comment (RalfJung) [reply packages/coq/coq.8.7.1/opam]: I did some googling but was unable to find information about "attributes" for opam dependencies. The word "attribute" also does not appear at <https://opam.ocaml.org/doc/Packaging.html>. Could you give me a pointer how I would add such an attribute?

Comment (jpdeplaix) [reply packages/coq/coq.8.7.1/opam]: In this doc it appears in the "Build depends" section.
In the dev manual they are referred as "dependency flags" on page 7: https://github.com/ocaml/opam/blob/master/doc/dev-manual/dev-manual.pdf

Comment (RalfJung) [reply packages/coq/coq.8.7.1/opam]: I see, thanks :)

**Comment by RalfJung** [comment]
I added the `build` flag as requested.

File: packages/coqide/coqide.8.7.1/opam [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+build: [
+  [
+    "./configure"
+    "-configdir" "%{lib}%/coq/config"
+    "-prefix" prefix
+    "-mandir" man
+    "-docdir" doc
+    "-libdir" "%{lib}%/coq"
+    "-datadir" "%{share}%/coq"
+    "-usecamlp5"
```
Comment (palmskog) [inline packages/coqide/coqide.8.7.1/opam]: Note that the `-usecamlp5` option is deprecated, since camlp5 is already a required dependency.

Comment (RalfJung) [reply packages/coqide/coqide.8.7.1/opam]: Good point, the option is not even listed any more in `./configure -help`. I removed it.

**Comment by jpdeplaix** [comment]
I'm going to go ahead and merge this but there are two problems that needs to be addressed sooner or later:
* ```md5sum --version``` is not valid on Alpine Linux. This needs to be fixed upstream in the next version.
* Next time please use ```"rm" "-f"``` to remove files. In case of a failure in the build phase, it displays several unnecessary warnings.

**Comment by RalfJung** [comment]
Thanks!

> md5sum --version is not valid on Alpine Linux. This needs to be fixed upstream in the next version.

What do you mean, "not valid"? Does Alpine not ship with `md5sum` any more?

**Comment by jpdeplaix** [comment]
> What do you mean, "not valid"? Does Alpine not ship with md5sum any more?

It does but ```--version``` is not a valid argument.

Issue reported on the Coq issue tracker https://github.com/coq/coq/issues/6456.

**Pull request merged**